### PR TITLE
PYIC-7069: repeat fraud check api tests

### PR DESCRIPTION
**Pull request by thebauSoftwire**
### What changed
Add api tests for repeat fraud check

### Why did it change
Increase coverage of api tests to cover key journeys

### Issue tracking

- [PYIC-7069](https://govukverify.atlassian.net/browse/PYIC-7069)

[PYIC-7069]: https://govukverify.atlassian.net/browse/PYIC-7069?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api-tests/src/steps/cri-steps.ts [hunk 1]
```diff
@@ -48,12 +50,18 @@ When(
       throw new Error("Last journey engine response was not a CRI response");
     }
 
+    const matchedRegex = /^(.*)-with-expiry$/.exec(scenario);
+    const nbf = matchedRegex === null ? undefined : EXPIRED_NBF;
+    const extractedScenario =
+      matchedRegex === null ? scenario : matchedRegex[1];
```
Comment (thebauSoftwire) [inline api-tests/src/steps/cri-steps.ts]: Wanted to re-use the stub data we already had so used regex to extract them but can get rid of this and just duplicate the data instead if this isn't as readable

File: api-tests/features/repeat-fraud-check.feature [hunk 2]
```diff
@@ -0,0 +1,69 @@
+Feature: Repeat fraud check journeys
+
+  @Build
+  Scenario: Fraud 6 Months Expiry + Name Update
```
Comment (sam803) [inline api-tests/features/repeat-fraud-check.feature]: You should be able to use a [Background](https://cucumber.io/docs/gherkin/reference/#background)  to reduce duplication :)

Comment (thebauSoftwire) [reply api-tests/features/repeat-fraud-check.feature]: oh nice thanks didn't know about this :)

File: api-tests/src/steps/cri-steps.ts [hunk 3]
```diff
@@ -14,6 +14,8 @@ import {
 } from "../types/internal-api.js";
 import { CriStubRequest } from "../types/cri-stub.js";
 
+const EXPIRED_NBF = 1658829758;
```
Comment (Joe-Edwards-GDS) [inline api-tests/src/steps/cri-steps.ts]: It's often nice to add a comment to values like this to explain them - e.g.
```
const EXPIRED_NBF = 1658829758; // 26/07/2022
```

File: api-tests/src/steps/cri-steps.ts (hunk 1 above)
Comment (Joe-Edwards-GDS) [reply api-tests/src/steps/cri-steps.ts]: This is ok, but if you use a step for this, you might get better autocomplete and it'll read nicer...

The simplest approach is to pull out the action as a method and define two steps that use it:
```
const submitCriDetails = async (world: World, scenario: string, nbf?: number): Promise<void> => {
  
};

When(
  "I submit {string} details to the CRI stub",
  async function (this: World, scenario: string): Promise<void> {
    await submitCriDetails(this, scenario);
  },
);

When(
  "I submit expired {string} details to the CRI stub",
  async function (this: World, scenario: string): Promise<void> {
    await submitCriDetails(this, scenario, EXPIRED_NBF);
  },
);
```

But a cleverer approach might be to switch to a regex step:
```
When(
  /I submit (expired )?'([\w-]+)' details to the CRI stub/,
  async function (this: World, expired: string, scenario: string): Promise<void> {
    await submitCriDetails(this, scenario, expired ? EXPIRED_NBF : undefined);
  },
);
```
(`expired` will have the value `expired ` if included, or undefined)

File: api-tests/features/repeat-fraud-check.feature [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    Then I get a 'fraud' CRI response
+    When I submit 'kenneth-changed-first-name-only-score-2-with-expiry' details to the CRI stub
+    Then I get a 'page-ipv-success' page response
+    When I submit a 'next' event
+    Then I get an OAuth response
+    When I use the OAuth response to get my identity
+    Then I get a 'P2' identity
+
+  @Build
+  Scenario: Fraud 6 Months Expiry + No Update
```
Comment (Joe-Edwards-GDS) [inline api-tests/features/repeat-fraud-check.feature]: A silly thing, but it feels more logical to me to put the simplest case first?

**Review by Joe-Edwards-GDS — COMMENTED** [review]
This looks good, but a couple of suggestions that might make it neater

File: api-tests/src/steps/cri-steps.ts (hunk 1 above)
Comment (thebauSoftwire) [reply api-tests/src/steps/cri-steps.ts]: I tried using a regex too but wasn't as nice as this thanks! :)

**Review by Joe-Edwards-GDS — APPROVED** [review]
@sam803 there'll probably be some overlap with your PR as well (particularly in the stub data) but hopefully not too hard to resolve

**Pull request merged**